### ISSUE TITLE
OMPython does not parse correctly expressions in array_subscripts

### ISSUE THREAD

**Issue by kataph** [OpenModelica/OMPython#231]
### Description

in modelica, array declarations allow subscripts to be expressions (consider e.g. this lines in the specifications:
```
array_subscripts :
"[" subscript { "," subscript } "]"
subscript :
":" | expression
```
)
Indeed, the openmodelica compiler can parse such things.
However, OMPython cannot deal with the ensuing outputs. 

### Steps to Reproduce

Minimum example:
```
import OMPython
omc = OMPython.OMCSessionZMQ()                                                                      
omc.sendExpression("loadString(\"model A Integer x[1+1,1]; end A;\")")
omc.sendExpression("getComponents(A)")
```

Actual result:
```
>>> import OMPython
>>> omc = OMPython.OMCSessionZMQ()                                                                      
>>> omc.sendExpression("loadString(\"model A Integer x[1+1,1]; end A;\")")
True
>>> omc.sendExpression("getComponents(A)")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Francesco\AppData\Local\Programs\Python\Python312\Lib\site-packages\OMPython\__init__.py", line 783, in sendExpression
    answer = OMTypedParser.parseString(result)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Francesco\AppData\Local\Programs\Python\Python312\Lib\site-packages\OMPython\OMTypedParser.py", line 120, in parseString
    res = omcGrammar.parseString(string)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Francesco\AppData\Local\Programs\Python\Python312\Lib\site-packages\pyparsing\util.py", line 256, in _inner
    return fn(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Francesco\AppData\Local\Programs\Python\Python312\Lib\site-packages\pyparsing\core.py", line 1200, in parse_string
    raise exc.with_traceback(None)
pyparsing.exceptions.ParseException: Expected end of text, found '{'  (at char 0), (line:1, col:1)
```
### Expected Behavior

That OMPython is able to parse correctly the raw output of the open modelica compiler getComponents scripting function, which in this case would be:
```
>> getComponents(A)
{{Integer, x, "", "public", false, false, false, false, "unspecified", "none", "unspecified", {1 + 1,1}}}
```
so, hypothetically, something like
```
(('Integer', 'x', '', 'public', False, False, False, False, 'unspecified', 'none', 'unspecified', (1+1, 1)),)
```

### Version and OS
 - Python Version: Python 3.12.2
 - OMPython Version:  3.6.0
 - OpenModelica Version: 'OpenModelica v1.24.0 (64-bit)'
 - OS: Windows 11 Home x64

**Comment by casella** [comment]
@arun3688 how do you see that? Is it easy to fix?

**Comment by arun3688** [comment]
@kataph the immediate solution would be to use parsed=False in sendExpression and then you do some conversion on the string by yourself in python see below

```
>>> import OMPython
>>> omc = OMPython.OMCSessionZMQ()                                                                      
>>> omc.sendExpression("loadString(\"model A Integer x[1+1,1]; end A;\")")
>>> omc.sendExpression("getComponents(A)", parsed=False) ## this returns the unparsed string 
```
the problem seems to be `x[1+1, 1]`

**Comment by kataph** [comment]
[arun3688](https://github.com/arun3688) 
Yes, the problem appears to be (at least) due arithmetic expressions. In the codebase I have to interact with there are various parameters that determine the dimensions of certain arrays by means of arithmetic operations, in particular summation. 
The solution I am using at the moment indeed takes the raw input and parses it in a cursorily way. However, it is not trivial for me to develop a parser that could parse any output of getComponent(...) with full generality.

**Comment by arun3688** [comment]
@kataph I added the rules to parser which can handle simple arithmetic operations in array dimensions, see the below examples

```
>>> import OMPython
>>> omc = OMPython.OMCSessionZMQ()
>>> omc.sendExpression("loadString(\"model A Integer x[5*3,1+7]; end A;\")")
True
>>> omc.sendExpression("getComponents(A)")
(('Integer', 'x', '', 'public', False, False, False, False, 'unspecified', 'none', 'unspecified', (15, 8)),) 
```
I hope you have simple arithmetic operations like in the example, if that is the case i will push the code soon

**Comment by arun3688** [comment]
@kataph the issue is fixed with this commit 317b944d32e98d824e66dc644b62331b20664fd8,  update your master or pip install via` python -m pip install -U https://github.com/OpenModelica/OMPython/archive/master.zip `

**Comment by arun3688** [comment]
> however, expressions with parameter-variables still fail. For instance:

Yes it will fail for sure, because we have to handle rules for those expression, i will try to fix it.

**Comment by arun3688** [comment]
@kataph i added rules to allow literals in array dimension, this should be fixed in this commit 6e0b61a21939f00b1921794f34abe2b9eba37dec, please pull the latest master and test it

**Comment by kataph** [comment]
Tested, it works! Many thanks.
